src/storage/recovery/recovery_manager.cpp

```diff
@@ -48,10 +48,21 @@ void RecoveryManager::WaitForRecoveryToFinish() {
 void RecoveryManager::RecoverFromLogs(const common::ManagedPointer<AbstractLogProvider> log_provider) {
   // Replay logs until the log provider no longer gives us logs
   while (true) {
-    if (replication_manager_ != DISABLED &&
-        log_provider->GetType() == AbstractLogProvider::LogProviderType::REPLICATION) {
-      auto rep_log_provider = log_provider.CastManagedPointerTo<ReplicationLogProvider>();
-      if (!rep_log_provider->NonBlockingHasMoreRecords()) break;
+    if (replication_manager_ != DISABLED && replication_manager_->IsReplica()) {
```

Comment:
I'm not sure if this matters, but the way this is set up it doesn't seem like it would be possible for a replica node to recover from a crash using the WAL.